core/Common.php

```diff
@@ -1021,8 +1024,16 @@ public static function getContinent($country)
      *            1 => array( ... ) // campaign keyword parameters
      * );
      */
-    public static function getCampaignParameters()
+    public static function getCampaignParameters(?int $idSite = null)
     {
+        $featureFlagManager = StaticContainer::get(FeatureFlagManager::class);
+        if ($featureFlagManager->isFeatureActive(PrivacyCompliance::class)) {
+            $cache = TrackerCache::getCacheWebsiteAttributes($idSite);
+            if ($cache[CampaignTrackingParametersDisabled::class] === true) {
```

Comment:
I think it's safer to null coalesce or use other mechanism to check if the array key really exists to prevent accessing an undefined array key.
I don't have an example of how it could happen that it would be tested before the tracker cache was populated, but checking other examples where tracker cache values are used they are always guarded by a check first.

```suggestion
            $cache = TrackerCache::getCacheWebsiteAttributes($idSite);
            $cacheKey = CampaignTrackingParametersDisabled::class;
            if (($cache[$cacheKey] ?? false) === true) {
```